src/parkapi_sources/converters/bahn_v2/mapper.py

```diff
@@ -6,41 +6,69 @@
 from datetime import datetime, timezone
 
 from parkapi_sources.models import StaticParkingSiteInput
+from parkapi_sources.models.enums import PurposeType
 
-from .validators import BahnParkingSiteCapacityType, BahnParkingSiteInput, NameContext
+from .validators import (
+    BahnParkingSiteCapacityType,
+    BahnParkingSiteInput,
+    NameContext,
+)
 
 
 class BahnMapper:
     @staticmethod
-    def map_static_parking_site(bahn_input: BahnParkingSiteInput) -> StaticParkingSiteInput:
+    def map_static_parking_site_car(bahn_input: BahnParkingSiteInput) -> StaticParkingSiteInput:
+        return BahnMapper._map_static_parking_site(
+            bahn_input, '-parking', PurposeType.CAR, BahnParkingSiteCapacityType.PARKING
+        )
+
+    @staticmethod
+    def map_static_parking_site_bike_locked(bahn_input: BahnParkingSiteInput) -> StaticParkingSiteInput:
+        return BahnMapper._map_static_parking_site(
+            bahn_input, '-bike-locked', PurposeType.BIKE, BahnParkingSiteCapacityType.BIKE_PARKING_LOCKED
+        )
+
+    @staticmethod
+    def map_static_parking_site_bike_open(bahn_input: BahnParkingSiteInput) -> StaticParkingSiteInput:
+        return BahnMapper._map_static_parking_site(
+            bahn_input, '-bike-open', PurposeType.BIKE, BahnParkingSiteCapacityType.BIKE_PARKING_OPEN
+        )
+
+    @staticmethod
+    def _map_static_parking_site(
+        bahn_input, uid_suffix: str, purpose: PurposeType, capacity_type: BahnParkingSiteCapacityType
+    ) -> StaticParkingSiteInput:
         static_parking_site_input = StaticParkingSiteInput(
-            uid=str(bahn_input.id),
+            uid=f'{bahn_input.id}{uid_suffix}',
+            group_uid=str(bahn_input.id),
             name=next(
                 iter(name_input.name for name_input in bahn_input.name if name_input.context == NameContext.NAME)
             ),
             lat=bahn_input.address.location.latitude,
             lon=bahn_input.address.location.longitude,
             operator_name=bahn_input.operator.name,
             address=f'{bahn_input.address.streetAndNumber}, {bahn_input.address.zip} {bahn_input.address.city}',
-            type=bahn_input.type.name.to_parking_site_type_input(),
+            type=next(
+                iter(
+                    item.to_bike_parking_site_type_input() for item in bahn_input.capacity if item.type == capacity_type
+                )
+            )
+            if capacity_type != BahnParkingSiteCapacityType.PARKING
```

Comment:
You can put this below, which would remove the inline if which is not in one line any more because of the length.